packages/ui/src/machines/authenticator/index.ts

```diff
@@ -313,14 +316,19 @@ export function createAuthenticatorMachine() {
         }),
       },
       guards: {
-        shouldRedirectToSignUp: (_, event): boolean => {
-          if (!event.data?.intent) return false;
-          return event.data.intent === 'confirmSignUp';
-        },
-        shouldRedirectToResetPassword: (_, event): boolean => {
-          if (!event.data?.intent) return false;
-          return event.data.intent === 'confirmPasswordReset';
-        },
+        // guards for initial states
+        shouldInitSignUp: (context) => context.config.initialState === 'signUp',
+        shouldInitResetPassword: (context) =>
```

Comment:
Not satisfied with names here -- maybe `isInitialStateSignUp`? `shouldSignUp`? open for thoughts...

Reply:
Do think `isInitialStateSignUp` may read cleaner

Reply:
Sounds good, adjusting!